app/lib/intervention.cjsx

```diff
@@ -0,0 +1,70 @@
+React = require 'react'
+InterventionMonitor = require './intervention-monitor'
+http = require 'http'
+experimentsClient = new (require './experiments-client')
+config = require './intervention-config'
+{getSessionID} = require '../lib/session'
+
+Intervention = React.createClass
+
+  contextTypes:
+    geordi: React.PropTypes.object
+    interventionMonitor: React.PropTypes.object
+
+  getInitialState: ->
+    latestFromSugar: null
+
+  cancelIntervention: (event) ->
+    # TODO cancel properly
+    @endIntervention(event)
+    @context.geordi?.logEvent
+      type: "cancelIntervention"
+
+  endIntervention: (event) ->
+    experiment_name = event.target.attributes.getNamedItem("data-experiment-name")?.value
+    user_id = event.target.attributes.getNamedItem("data-user-id")?.value
+    session_id = event.target.attributes.getNamedItem("data-session-id")?.value
+    intervention_id = event.target.attributes.getNamedItem("data-next-event")?.value
+    type = "intervention"
+    experimentsClient.postDataToExperimentServer @context.interventionMonitor,
+                                                 @context.geordi,
+                                                 experiment_name,
+                                                 user_id,
+                                                 session_id,
+                                                 type,
+                                                 intervention_id
+
+  render: ->
+    if @context.interventionMonitor?.latestFromSugar
+      experiment_name = @context.interventionMonitor?.latestFromSugar["experiment_name"]
+      user_id = @props.user.id
+      session_id = getSessionID()
+      intervention_id = @context.interventionMonitor?.latestFromSugar["next_event"]
+      intervention_details = experimentsClient.constructInterventionFromSugarData @context.interventionMonitor?.latestFromSugar
```

Comment:
I don't think any of these need to be in the render. Instead I would suggest using a combination of props/state and use them directly in your handlers.